### PR TITLE
obs-studio-plugins.droidcam-obs: init at 1.6.1

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Description of changes

I've used master for a while and it's worked very nicely for me.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SebTM — APPROVED** [review]

**Comment by SebTM** [comment]
Build and tested against unstable and with android-phone, works for me :)

**Comment by ulrikstrid** [comment]
I've been using this since I opened the PR without issues. Used a version of it pointing at master before, also without issues.

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+, libjpeg
+, libimobiledevice
+, libusbmuxd
+, libplist
+}:
+
+stdenv.mkDerivation rec {
+  pname = "droidcam-obs";
+  version = "2.0.1";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix:15]:
```suggestion
  version = "2.0.1";

  src = fetchFromGitHub {
```

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    owner = "dev47apps";
+    repo = "droidcam-obs-plugin";
+    rev = version;
+    sha256 = "sha256-oaw/mq4WCQMlf3sv9WtNlv9J9rm79xnqDwKzHtyFW50=";
+  };
+
+
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix:22]:
```suggestion

```

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    "JPEG_DIR=${libjpeg.dev}"
+    "JPEG_LIB=${libjpeg.out}/lib"
+    "IMOBILEDEV_DIR=${libimobiledevice}"
+  ];
+
+  installPhase = ''
+    mkdir -p $out/share/obs/obs-plugins/droidcam-obs
+    mkdir -p $out/lib/obs-plugins
+    cp build/droidcam-obs.so $out/lib/obs-plugins
+    cp build/droidcam-obs.so $out/share/obs/obs-plugins/droidcam-obs
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix:55]: Please use symlinks instead

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      --replace "-limobiledevice" "-limobiledevice-1.0" \
+      --replace "-I/usr/include/obs" "-I${obs-studio}/include/obs" \
+      --replace "-I/usr/include/ffmpeg" "-I${ffmpeg}/include"
+  '';
+
+  preBuild = ''
+    mkdir ./build
+  '';
+
+  propagatedBuildInputs = [
```
Comment (raphaelr) [inline pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix:34]: These should just be normal `buildInputs`, instead of propagated ones

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    libjpeg
+    libimobiledevice
+    libimobiledevice.dev
```
Comment (raphaelr) [inline pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix:37]:
```suggestion
    libimobiledevice
```

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    libusbmuxd
+    libplist
+    obs-studio
+    ffmpeg
+  ];
+
+  makeFlags = [
+    "ALLOW_STATIC=no"
+    "JPEG_DIR=${libjpeg.dev}"
+    "JPEG_LIB=${libjpeg.out}/lib"
```
Comment (raphaelr) [inline pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix:47]:
```suggestion
    "JPEG_DIR=${lib.getDev libjpeg}"
    "JPEG_LIB=${lib.getLib libjpeg}/lib"
```
lib.getDev automatically falls back to the `out` output if the `dev` one does not exist. Same for lib.getLib.

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    cp build/droidcam-obs.so $out/share/obs/obs-plugins/droidcam-obs
+    cp -R ./data/locale $out/share/obs/obs-plugins/droidcam-obs/locale
+  '';
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "DroidCam OBS";
+    homepage = "https://github.com/dev47apps/droidcam-obs-plugin";
+    license = licenses.gpl2;
```
Comment (raphaelr) [inline pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix:64]:
```suggestion
    license = licenses.gpl2Plus;
```
gpl2 is legacy, use either gpl2Plus or gpl2Only.

**Review by raphaelr — CHANGES_REQUESTED** [review]
Please change the `droidcam-obs` in the commit message and PR title to `obs-studio-plugins.droidcam-obs`, so that ofborg will attempt to build it.

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix (hunk 3 above)
Comment (ulrikstrid) [reply pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix]: For both or just 1?

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix (hunk 6 above)
Comment (ulrikstrid) [reply pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix]: Did not know about these functions, thank you

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  ];
+
+  makeFlags = [
+    "ALLOW_STATIC=no"
+    "JPEG_DIR=${lib.getDev libjpeg}"
+    "JPEG_LIB=${lib.getLib libjpeg}/lib"
+    "IMOBILEDEV_DIR=${libimobiledevice}"
+  ];
+
+  installPhase = ''
```
Comment (raphaelr) [inline pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix:50]: add `runHook preInstall` at the beginning, and `runHook postInstall` at the end.

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    mkdir -p $out/share/obs/obs-plugins/droidcam-obs
+    mkdir -p $out/lib/obs-plugins
+    cp build/droidcam-obs.so $out/lib/obs-plugins
+    ln -s build/droidcam-obs.so $out/share/obs/obs-plugins/droidcam-obs
```
Comment (raphaelr) [inline pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix:54]: Is this file required? The `$out/lib/droidcam-obs.so` file should be enough for OBS to load the plugin.

If it is required: This is a dangling symlink, as the `build/` directory only exists during the build. The line should instead be `ln -s
$out/lib/obs-plugins/droidcam-obs.so $out/share/obs/obs-plugins/droidcam-obs`

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix (hunk 8 above)
Comment (ulrikstrid) [reply pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix:50]: Done

File: pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix (hunk 9 above)
Comment (ulrikstrid) [reply pkgs/applications/video/obs-studio/plugins/droidcam-obs/default.nix]: You are correct, removed the second line

**Comment by ulrikstrid** [comment]
Updated from the comments and tested to make sure it still works :ok_hand:

**Review by raphaelr — APPROVED** [review]
